src/common/cockpitsystem.c

```diff
@@ -90,6 +94,48 @@ cockpit_system_load_os_release (void)
   return result;
 }
 
+JsonObject *
+cockpit_system_load_os_release_to_json (void)
+{
+  JsonObject *block = NULL;
+  gint i;
+  const gchar *value;
+
+  GHashTable *os_release = cockpit_system_load_os_release();
+
+  if (os_release)
+   {
+     block = json_object_new ();
+     for (i = 0; i < G_N_ELEMENTS (os_release_fields); i++)
+       {
+         value = g_hash_table_lookup (os_release, os_release_fields[i]);
+         if (value)
+           json_object_set_string_member (block, os_release_fields[i], value);
+       }
+   }
+
+  g_hash_table_unref (os_release);
+  return block;
+}
+
+GHashTable *
+cockpit_system_load_os_release_from_json (JsonObject *object)
+{
+  gint i;
+  GHashTable *os_release = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+
+  for (i = 0; i < G_N_ELEMENTS (os_release_fields); i++)
```

Comment:
I think this function should decode anything that is encoded, and not be limited to os_release_fields. As a general principle between the various cockpit parts.